copy/copy.go

```diff
@@ -124,6 +125,12 @@ type copier struct {
 	downloadForeignLayers bool
 }
 
+// hasStorageDestination returns true if the copier's `dest` points to a
+// containers-storage destination.
+func (c *copier) hasStorageDestination() bool {
+	return c.dest.Reference().Transport().Name() == internalStorage.Transport.Name()
```

Comment:
Also, IIRC the blob cache or some other components may wrap transports to modify behavior, but return the original names — in that case this name check could pass but the later type check would fail instead of going down the compatibility path.

Reply:
Very good point, will change.